packages/fast-components-react-base/src/viewport-positioner/viewport-positioner.tsx

```diff
@@ -0,0 +1,1011 @@
+                        this.props.managedClasses,
+                        "viewportPositioner__horizontalInset",
+                        ""
+                    )
+                );
+                break;
+
+            case ViewportPositionerHorizontalPositionLabel.insetRight:
+                classNames = classNames.concat(
+                    " ",
+                    get(this.props.managedClasses, "viewportPositioner__right", "")
+                );
+                classNames = classNames.concat(
+                    " ",
+                    get(
+                        this.props.managedClasses,
+                        "viewportPositioner__horizontalInset",
+                        ""
+                    )
+                );
+                break;
+
+            case ViewportPositionerHorizontalPositionLabel.right:
+                classNames = classNames.concat(
+                    " ",
+                    get(this.props.managedClasses, "viewportPositioner__right", "")
+                );
+                break;
+        }
+
+        switch (this.state.currentVerticalPosition) {
+            case ViewportPositionerVerticalPositionLabel.top:
+                classNames = classNames.concat(
+                    " ",
+                    get(this.props.managedClasses, "viewportPositioner__top", "")
+                );
+                break;
+
+            case ViewportPositionerVerticalPositionLabel.insetTop:
+                classNames = classNames.concat(
+                    " ",
+                    get(this.props.managedClasses, "viewportPositioner__top", "")
+                );
+                classNames = classNames.concat(
+                    " ",
+                    get(
+                        this.props.managedClasses,
+                        "viewportPositioner__verticalInset",
+                        ""
+                    )
+                );
+                break;
+
+            case ViewportPositionerVerticalPositionLabel.insetBottom:
+                classNames = classNames.concat(
+                    " ",
+                    get(this.props.managedClasses, "viewportPositioner__bottom", "")
+                );
+                classNames = classNames.concat(
+                    " ",
+                    get(
+                        this.props.managedClasses,
+                        "viewportPositioner__verticalInset",
+                        ""
+                    )
+                );
+                break;
+
+            case ViewportPositionerVerticalPositionLabel.bottom:
+                classNames = classNames.concat(
+                    " ",
+                    get(this.props.managedClasses, "viewportPositioner__bottom", "")
+                );
+                break;
+        }
+
+        return super.generateClassNames(classNames);
+    }
+
+    /**
+     *  gets the CSS classes to be programmatically applied to the component
+     */
+    private getPositioningStyles = (): {} => {
```

Comment:
can you type this as a Partial?

Reply:
A partial what?  This is a collection of arbitrary css attributes, not props or state so it would not be a part of anything that I have defined.

Reply:
"React.CSSProperties" worked